src/plot.cpp

```diff
@@ -980,16 +1019,552 @@ RGBColor random_color(void)
+
+      if (vert < pixels_[1]) {
+
+        for (int horiz = 0; horiz < pixels_[0]; ++horiz) {
+
+          // Generate the starting position/direction of the ray
+          if (orthographic_width_ == 0.0) { // perspective projection
+            double this_phi = -horiz_fov_radians / 2.0 + dphi * horiz;
+            double this_mu = -vert_fov_radians / 2.0 + dmu * vert + M_PI / 2.0;
+            Direction camera_local_vec;
+            camera_local_vec.x = std::cos(this_phi) * std::sin(this_mu);
+            camera_local_vec.y = std::sin(this_phi) * std::sin(this_mu);
+            camera_local_vec.z = std::cos(this_mu);
+            s.u = camera_local_vec.rotate(camera_to_model);
+          } else { // orthographic projection
+            s.u = looking_direction;
+
+            double x_pix_coord = (static_cast<double>(horiz) - p0 / 2.0) / p0;
+            double y_pix_coord = (static_cast<double>(vert) - p1 / 2.0) / p0;
+            s.r = camera_position_ +
+                  cam_yaxis * x_pix_coord * orthographic_width_ +
+                  cam_zaxis * y_pix_coord * orthographic_width_;
+          }
+
+          p.from_source(&s); // put particle at camera
+          bool hitsomething = false;
+          bool intersection_found = true;
+          int loop_counter = 0;
+          const int max_intersections = 1000000;
+
+          this_line_segments[tid][horiz].clear();
+
+          int first_surface =
+            -1; // surface first passed when entering the model
+          bool first_inside_model = true; // false after entering the model
+          while (intersection_found) {
+            bool inside_cell = exhaustive_find_cell(p);
+            if (inside_cell) {
+
+              // This allows drawing wireframes with surface intersection
+              // edges on the model boundary for the same cell.
+              if (first_inside_model) {
+                this_line_segments[tid][horiz].emplace_back(
+                  color_by_ == PlotColorBy::mats
+                    ? p.material()
+                    : p.coord(p.n_coord() - 1).cell,
+                  0.0, first_surface);
+                first_inside_model = false;
+              }
+
+              hitsomething = true;
+              intersection_found = true;
+              auto dist = distance_to_boundary(p);
+              this_line_segments[tid][horiz].emplace_back(
+                color_by_ == PlotColorBy::mats ? p.material()
+                                               : p.coord(p.n_coord() - 1).cell,
+                dist.distance, std::abs(dist.surface_index));
+
+              // Advance particle
+              for (int lev = 0; lev < p.n_coord(); ++lev) {
+                p.coord(lev).r += dist.distance * p.coord(lev).u;
+              }
+              p.surface() = dist.surface_index;
+              p.n_coord_last() = p.n_coord();
+              p.n_coord() = dist.coord_level;
+              if (dist.lattice_translation[0] != 0 ||
```

Comment:
I think you can also check the lowest coordinate of particle for `lattice >= 0`.

Reply:
What if the particle is in a lattice, but is not crossing a lattice boundary? Wouldn't the lowest coordinate then have the lattice variable defined, but it wouldn't be appropriate to call cross_lattice?

Reply:
Oh yeah totally. My bad!